fix: Always patch PayPal ordernumber

aragon999 opened the pull request:
A customer of ours wanted to have a prefix but with PayPal Plus disabled.
In this case the order number is not patched and the standard Shopware order number is used, see:
https://github.com/shopwareLabs/SwagPayPal/blob/2.2.5/src/Checkout/Payment/Handler/PayPalHandler.php#L144

I guess this PR is not complete, since there are probably cases when the order number should not be patched, looking forward to some feedback.

mitelg commented:
hey @aragon999 
thanks for your contribution. I guess, you might already know that changing a variable to a simple `true`, where there was quite the logic before, is certainly not the solution :wink: 

The reason that the order number must be patched for those special cases is, that the payment is already created on PayPal site before we have the order in Shopware. The PayPal payment must be created before, because of the fancy Javascript things on the confirm page (Plus iFrame, Smart Payment Buttons) or because of express checkout. (for this the [OrderFromCartBuilder](https://github.com/shopwareLabs/SwagPayPal/blob/2.2.5/src/OrdersApi/Builder/OrderFromCartBuilder.php) is used)

If Plus and Smart Payment Buttons are deactivated we have a simple redirect to the PayPal page. In this case, the Shopware order is already present and we can send the order number directly to PayPal (see [OrderFromOrderBuilder](https://github.com/shopwareLabs/SwagPayPal/blob/2.2.5/src/OrdersApi/Builder/OrderFromOrderBuilder.php#L97-L100))

So in the end, the order number should always be sent to PayPal. But looking at the OrderFromOrderBuilder we already have the problem :smile: 
The [whole logic, ](https://github.com/shopwareLabs/SwagPayPal/blob/2.2.5/src/Checkout/Payment/Handler/PayPalHandler.php#L147-L152) about sending the order number or not and the prefix setting, is missing.

So this is the place, where you should start :blush: :muscle:

aragon999 commented:
Actually that was my intention to get some feedback :-) And the fix worked for the customer for now.

I just updated the PR, as I think it should be correct.

I also added a test, but was not able to get it running locally (probably because of my specific setup). 
But if needed I can get it running tomorrow.

mitelg commented:
looks good for me :+1:

aragon999 commented:
Btw. since we are using Shopware 6.3.5 it would be good if the fix is backported to the `2.2` branch (I could not find it here on Github), but I guess you would have taken care about that anyways.

mitelg commented:
merged with https://github.com/shopwareLabs/SwagPayPal/commit/4ef71ba3ff71881ca54d157845c0d397e2275799 into the SW6.3 version. the merge on master will follow on monday

thanks for your contribution @aragon999   👍🎉💙
